File: PIProduct/entity/Produit.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package gestionproduit.entity;

import java.io.File;

/**
 *
 * @author wiemj
 */
public class Produit {
 
    private int idProduit;
    private int idOwer;
    private String categorieProduit;
    private String nomProduit;
    private String descriptionProduit;
    private String photoProduit;
    private String videoProduit;
    private float prixProduit;
    private int stockProduit;
    private File Image;
    public static int id_pModifier;

    public Produit() {
    }

    

 

    public File getImage() {
        return Image;
    }

    public void setImage(File Image) {
        this.Image = Image;
    }

        public Produit( String categorieProduit, int idOwner, String descriptionProduit,  float prixProduit, int stockProduit,  String nomProduit, String photoProduit) {        this.nomProduit = nomProduit;

        this.categorieProduit = categorieProduit;
        this.descriptionProduit = descriptionProduit;
        this.photoProduit=photoProduit;
        this.prixProduit = prixProduit;
        this.stockProduit = stockProduit;
        this.idOwer=idOwner;
    }
     public Produit(int idProduit ,String nomProduit , String descriptionProduit, String photoProduit,  float prixProduit, int stockProduit,  String categorieProduit
) {        this.nomProduit = nomProduit;

        this.categorieProduit = categorieProduit;
        this.descriptionProduit = descriptionProduit;
        this.photoProduit = photoProduit;
        this.prixProduit = prixProduit;
        this.stockProduit = stockProduit;
    }
     public Produit(String categorieProduit, int idOwer, String descriptionProduit,  float prixProduit, int stockProduit,  String nomProduit
) {        
        this.idOwer =idOwer;
        this.nomProduit = nomProduit;
         this.categorieProduit = categorieProduit;
        this.descriptionProduit = descriptionProduit;
        this.photoProduit = photoProduit;
        this.prixProduit = prixProduit;
        this.stockProduit = stockProduit;
    }
    public Produit( String categorieProduit, String descriptionProduit, String photoProduit, String videoProduit, float prixProduit, int stockProduit,  String nomProduit
) {        this.nomProduit = nomProduit;

        this.categorieProduit = categorieProduit;
        this.descriptionProduit = descriptionProduit;
        this.photoProduit = photoProduit;
        this.videoProduit = videoProduit;
        this.prixProduit = prixProduit;
        this.stockProduit = stockProduit;
    }
    public Produit( String categorieProduit, String descriptionProduit, String photoProduit,  float prixProduit, int stockProduit,  String nomProduit
) {        this.nomProduit = nomProduit;

        this.categorieProduit = categorieProduit;
        this.descriptionProduit = descriptionProduit;
        this.photoProduit = photoProduit;
        this.prixProduit = prixProduit;
        this.stockProduit = stockProduit;
    }
    public Produit(String categorieProduit, String descriptionProduit, String photoProduit, String videoProduit) {
        this.categorieProduit = categorieProduit;
        this.descriptionProduit = descriptionProduit;
        this.photoProduit = photoProduit;
        this.videoProduit = videoProduit;
    }
 public Produit(String descriptionProduit, String photoProduit, String videoProduit) {
        this.descriptionProduit = descriptionProduit;
        this.photoProduit = photoProduit;
        this.videoProduit = videoProduit;
    }
   

    public Produit(String nom_produit, String description, String photo, float prix_produit, int stock_produit, String categorie_produit, String video) {
 this.nomProduit = nom_produit;
        this.descriptionProduit = description;
        this.photoProduit = photo;
        this.categorieProduit = categorie_produit;
        this.prixProduit = prix_produit;
        this.stockProduit = stock_produit;
        this.videoProduit= video;
    }

    public Produit(int id_produit, String nom_produit, String description, String photo, float prix_produit, int stock_produit, String categorie_produit, String video) {
        this.idProduit= id_produit;
        this.nomProduit = nom_produit;
        this.descriptionProduit = description;
        this.photoProduit = photo;
        this.categorieProduit = categorie_produit;
        this.prixProduit = prix_produit;
        this.stockProduit = stock_produit;
        this.videoProduit= video;    }

   


  

   

  

    public int getIdProduit() {
        return idProduit;
    }

    public void setIdProduit(int idProduit) {
        this.idProduit = idProduit;
    }

    public String getCategorieProduit() {
        return categorieProduit;
    }

    public void setCategorieProduit(String categorieProduit) {
        this.categorieProduit = categorieProduit;
    }

    public String getNomProduit() {
        return nomProduit;
    }

    public void setNomProduit(String nomProduit) {
        this.nomProduit = nomProduit;
    }

    public String getDescriptionProduit() {
        return descriptionProduit;
    }

    public void setDescriptionProduit(String descriptionProduit) {
        this.descriptionProduit = descriptionProduit;
    }

    public String getPhotoProduit() {
        return photoProduit;
    }

    public void setPhotoProduit(String photoProduit) {
        this.photoProduit = photoProduit;
    }

    public String getVideoProduit() {
        return videoProduit;
    }

    public void setVideoProduit(String videoProduit) {
        this.videoProduit = videoProduit;
    }

    public float getPrixProduit() {
        return prixProduit;
    }

    public void setPrixProduit(float prixProduit) {
        this.prixProduit = prixProduit;
    }

    public int getStockProduit() {
        return stockProduit;
    }

    public void setStockProduit(int stockProduit) {
        this.stockProduit = stockProduit;
    }

    public int getIdOwer() {
        return idOwer;
    }

    public void setIdOwer(int idOwer) {
        this.idOwer = idOwer;
    }

    public static int getId_pModifier() {
        return id_pModifier;
    }

    public static void setId_pModifier(int id_pModifier) {
        Produit.id_pModifier = id_pModifier;
    }

    
    
    
}

File: src/Entity/Produit.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * Produit
 *
 * @ORM\Table(name="produit")
 * @ORM\Entity
 */
class Produit
{
    /**
     * @var int
     *
     * @ORM\Column(name="id_produit", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $idProduit;

    /**
     * @var int
     *
     * @ORM\Column(name="id_owner", type="integer", nullable=false)
     */
    private $idOwner;

    /**
     * @var string
     *
     * @ORM\Column(name="nom_produit", type="string", length=255, nullable=false)
     */
    private $nomProduit;

    /**
     * @var string
     *
     * @ORM\Column(name="categorie_produit", type="string", length=255, nullable=false)
     */
    private $categorieProduit;

    /**
     * @var float
     *
     * @ORM\Column(name="prix_produit", type="float", precision=10, scale=0, nullable=false)
     */
    private $prixProduit;

    /**
     * @var int
     *
     * @ORM\Column(name="stock_produit", type="integer", nullable=false)
     */
    private $stockProduit;

    /**
     * @var string
     *
     * @ORM\Column(name="description", type="string", length=255, nullable=false)
     */
    private $description;

    /**
     * @var string
     *
     * @ORM\Column(name="photo", type="blob", length=0, nullable=false)
     */
    private $photo;

    public function getIdProduit(): ?int
    {
        return $this->idProduit;
    }

    public function getIdOwner(): ?int
    {
        return $this->idOwner;
    }

    public function setIdOwner(int $idOwner): self
    {
        $this->idOwner = $idOwner;

        return $this;
    }

    public function getNomProduit(): ?string
    {
        return $this->nomProduit;
    }

    public function setNomProduit(string $nomProduit): self
    {
        $this->nomProduit = $nomProduit;

        return $this;
    }

    public function getCategorieProduit(): ?string
    {
        return $this->categorieProduit;
    }

    public function setCategorieProduit(string $categorieProduit): self
    {
        $this->categorieProduit = $categorieProduit;

        return $this;
    }

    public function getPrixProduit(): ?float
    {
        return $this->prixProduit;
    }

    public function setPrixProduit(float $prixProduit): self
    {
        $this->prixProduit = $prixProduit;

        return $this;
    }

    public function getStockProduit(): ?int
    {
        return $this->stockProduit;
    }

    public function setStockProduit(int $stockProduit): self
    {
        $this->stockProduit = $stockProduit;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }

    public function getPhoto()
    {
        return $this->photo;
    }

    public function setPhoto($photo): self
    {
        $this->photo = $photo;

        return $this;
    }


}
